cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -85,7 +85,7 @@ public void setRepository(FullHttpRequest request, HttpResponder responder,
                          GSON.toJson(
                            new SetRepositoryResponse(
                              String.format("Updated repository configuration for namespace '%s'.", namespaceId))));
-    } catch (InvalidRepositoryConfigException e) {
+    } catch (RepositoryConfigValidationException e) {
       responder.sendJson(HttpResponseStatus.BAD_REQUEST, GSON.toJson(new SetRepositoryResponse(e)));
```

Comment:
In `HttpExceptionHandler`, cause.getMessage string is added as the response body, while here a Json object is being returned with a list of failures. Since this was part of another PR, I can create a TODO to change this if you feel its required.

cc @de-lan

Reply:
Thanks! Pls add a TODO here, I'll fix this in a following PR

Reply:
fyi, I've added the changes in draft https://github.com/cdapio/cdap/pull/14905. Will rebase once this PR gets merged